tests/test_integration.py

```diff
@@ -13,7 +13,7 @@
 
 from conftest import state_backends
 
-pldevices = [("qiskit.aer", aer.Aer), ("qiskit.basicaer", qiskit.BasicAer)]
+pldevices = [("qiskit.aer", aer.AerProvider()), ("qiskit.basicaer", aer.AerProvider())]
```

Comment:
Isn't the provider for `basicaer` should be the new [BasicProvider](https://docs.quantum.ibm.com/api/qiskit/qiskit.providers.basic_provider.BasicProvider)? 🤔

Reply:
That didn't work out the door, and this allows us to continue using the existing backend (OpenQASM). I don't see a deprecation warning locally for OpenQASM as long as it's coming from the `AerProvider`, and I've updated to 0.46, but it's possible I've messed something up somewhere. This is a minimal change to make tests pass, not a decision on what do with the Aer device for the next release.

If Tom decides at standup that it's a clear choice and we just want to change to `BasicProvider`, then I'm happy to implement it here, might as well get it out of the way. But if it needs any further investigation/thought, I'd prefer we do it separately, as it's not a high priority, and the tests failing is blocking everything.